### PR TITLE
expose flk devshell module

### PR DESCRIPTION
**Pull request by blaggacao**
a more useful attempt than #39 for #38

### CODE REVIEW

File: shells/flk/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }: {
+  config = let
+    nixpkgs = import ../../compat/nixpkgs.nix;
```
Comment (blaggacao) [inline shells/flk/default.nix:3]: This looks suspicious. A way to avoid `./compat` here?

**Review by blaggacao — COMMENTED** [review]
I wonder...

File: shells/flk/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    configs = "${toString ../..}#nixosConfigurations";
```
Comment (blaggacao) [inline shells/flk/default.nix:4]: Can this be pulled in independent of this repo layout?

File: shells/flk/default.nix (hunk 1 above)
Comment (nrdxp) [reply shells/flk/default.nix:3]: The compat import was devised in order to ensure new users, without the experimental `nix-command`, can get their shell inputs from the same nixpkgs that the flake itself is tracking. This ensures that they have updated scripts, such as `nixos-install` that are flake compatible.

Once nix 3 is release, we can probably remove it.

**Comment by nrdxp** [comment]
What command can I run to access the shells. I've tried `nix develop ".#devshellModules.flk" with no success.

**Comment by blaggacao** [comment]
image

`.#devshellModules.flk` would be for somebody else to compose:

```nix
{
  inputs.flake-utils.url = "github:numtide/flake-utils";
  inputs.devshell.url = "github:numtide/devshell";
  inputs.nixflk.uri = "github:nrdxp";

  outputs = { self, flake-utils, devshell, nixflk, nixpkgs }:
    flake-utils.lib.simpleFlake {
      inherit nixpkgs;
      name = "my-project";
      preOverlays = [ (import devshell.overlay) ];
      shell = { pkgs }:
        pkgs.mkDevShell {
          imports  = [ (import nixflk.devshellModules.flk) ];
        };
    };
}
```

File: shells/flk/default.nix (hunk 1 above)
Comment (blaggacao) [reply shells/flk/default.nix:3]: I see, I tried to keep the bulk of this logic back in `shell.nix`. We should find a way to keep this module clean of compat or other repo local references before acceptation.

**Comment by blaggacao** [comment]
You question promoted this thought: https://github.com/numtide/devshell/issues/52#issuecomment-752264795

**Comment by blaggacao** [comment]
This should be (theoretically) good to go, I've rebase.

Setting a precedent for sharing devshells according to discussion over at https://github.com/numtide/devshell/issues/52#issuecomment-752065359
```console
$ nix flake show
git+file:///home/blaggacao/ghq/github.com/nrdxp/nixflk
...
├───devshellModules: unknown
...
```

And `nix develop` works as expected

```console
$ nix develop
🔨 Welcome to flk

[general commands]

  hooks       - install git hooks
  menu        - prints this menu

[linters]

  nixpkgs-fmt - Nix code formatter for nixpkgs

[main]

  flk         - Build, deploy, and install nixflk
  nix         - Powerful package manager that makes package management reliable and reproducible

[servers]

  grip        - Preview GitHub Markdown files like Readme locally before committing them

[vcs]

  git         - Distributed version control system
```

**Comment by blaggacao** [comment]
@zimbatm would you mind giving your feedback? (mainly to make sure this attempt on devshell sharing is +- inline with your ideas or even plans, if any)

**Comment by zimbatm** [comment]
The idea of shareable modules was for projects that want to provide extensions to the devshell. It seems that nixflk is a project template, so I would really keep things as simple as possible. If there are re-usable bits, break them out into a different project.

**Comment by blaggacao** [comment]
Now, with the new [devshell module system](https://github.com/numtide/devshell/tree/master/modules) in place, the impetus of this PR merits revisiting from a different angle some time in the near future on behalf of nixflk.

So I'm closing this for now and might think over it again in the coming days (hopefully, I'll find some time).

**Pull request closed without merging**